src/Analyzers/VisualBasic/CodeFixes/IncludeParenthesesForMethodInvocations/VisualBasicIncludeParenthesesForMethodInvocationsCodeFixProvider.vb

```diff
@@ -0,0 +1,73 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Diagnostics.CodeAnalysis
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.CodeActions
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.IncludeParenthesesForMethodInvocations
+    <ExportCodeFixProvider(LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicIncludeParenthesesForMethodInvocationsCodeFixProvider
+        Inherits SyntaxEditorBasedCodeFixProvider
+
+        <ImportingConstructor>
+        <SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification:="Used in test code: https://github.com/dotnet/roslyn/issues/42814")>
+        Public Sub New()
+        End Sub
+
+        Public Overrides ReadOnly Property FixableDiagnosticIds As ImmutableArray(Of String) = ImmutableArray.Create(IDEDiagnosticIds.IncludeParenthesesForMethodInvocationsDiagnosticId)
+
+        Friend Overrides ReadOnly Property CodeFixCategory As CodeFixCategory = CodeFixCategory.CodeStyle
+
+        Public Overrides Async Function RegisterCodeFixesAsync(context As CodeFixContext) As Task
+            Dim root = Await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(False)
+            For Each diagnostic In context.Diagnostics
+                Dim node = root.FindNode(diagnostic.Location.SourceSpan)
```

Comment:
i would prefer you use .AdditionalLocations, not Location.  The latter specifies what should be squiggled/faded.  The former allows you to providde the exact span of hte node to find.

Reply:
The next lines are based on the intentionally usage of "Location".
The reported additional location is always "InvocationExpressionSyntax", but the "Location" is either "ArgumentList" or "InvocationExpressionSyntax".

shouldn't I rely on that?

Reply:
please do not.  The location is the part we want to squiggle/fade, etc.  it may be tweaked.  teh .AdditinoalLocations are the spans you should use to retrieve nodes in teh fixer.  that's our pattern.